mcp/common/src/main/java/org/springframework/ai/mcp/SyncMcpToolCallbackProvider.java

```diff
@@ -133,7 +131,8 @@ public ToolCallback[] getToolCallbacks() {
 			.flatMap(mcpClient -> mcpClient.listTools()
 				.tools()
 				.stream()
-				.filter(tool -> this.toolFilter.test(mcpClient, tool))
+				.filter(tool -> this.toolFilter.test(new McpClientMetadata(mcpClient.getClientCapabilities(),
+						mcpClient.getClientInfo(), mcpClient.initialize()), tool))
```

Comment:
The use of client.initialize() will be replaced once we have access to the client.getCurrentInitializationResult() from https://github.com/modelcontextprotocol/java-sdk/pull/424